llvm/lib/ObjectYAML/GOFFEmitter.cpp

```diff
@@ -245,6 +248,57 @@ void GOFFState::writeHeader(GOFFYAML::FileHeader &FileHdr) {
   }
 }
 
+void GOFFState::writeSymbol(GOFFYAML::Symbol Sym) {
+  if (Sym.ID != SymbolID + 1)
+    reportError("symbol IDs not monotonic " + Sym.Name);
```

Comment:
I thought I raised this point before: in general, yaml2obj should be permissive in what it accepts as input, as one of the main purposes of yaml2obj is to craft inputs for tests. To test edge cases in file format handling, you need to be able to craft inputs that don't follow the rules of the file format. In other words, if you don't need symbol IDs (in this case) to be monotonic to be able to physically write things, don't enforce it.